Core/WebCacheManager.swift

```diff
@@ -64,7 +64,10 @@ public class WebCacheManager {
                 cookieStorage.setCookie(cookie)
                 
             }
-            WebCacheManager.dataStore.removeData(ofTypes: self.allDataTypes, modifiedSince: Date.distantPast) {}
+            
+            DispatchQueue.main.async {
```

Comment:
@subsymbolic This fixes the crash.  The code below wasn't running on the UI thread but only seemed to cause the crash when you used the fire button on the tabs screen after a couple of attempts. Easy to replicate it though.

Reply:
My questions is why? I see a timing change, but don't yet understand why it would fix the crash. My concern is it might mask it/move the timing rather than fix the root cause. Happy to zoom to discuss.

Reply:
When I debugged the crash the logs told me explicitly that that code needs to be run on the UI thread and that it wasn't.

Reply:
Ahh thanks @brindy, that makes sense, testing it now.